src/gen.rs

```diff
@@ -1,18 +1,238 @@
-use crate::range::Size;
+use crate::random;
+use crate::random::Random;
+use crate::range;
+use crate::range::{Range, Size};
+use crate::seed;
 use crate::seed::Seed;
+use crate::shrink;
+use crate::tree;
 use crate::tree::Tree;
+use num::{FromPrimitive, Integer, ToPrimitive};
+use std::rc::Rc;
 
-pub struct Gen<'a, A> {
-    #[allow(dead_code)]
-    un_gen: Fn(Size, Seed) -> Tree<'a, Option<A>>,
+#[derive(Clone)]
+pub struct Gen<'a, A>(#[allow(dead_code)] Random<'a, Tree<'a, A>>);
+
+// TODO: Would be handy to have From and Into traits implemented for
+// this to avoid a lot of the boilerplate.
+pub fn from_random<'a, A>(r: Random<'a, Tree<'a, A>>) -> Gen<'a, A> {
+    Gen(r)
+}
+
+pub fn to_random<'a, A>(g: Gen<'a, A>) -> Random<Tree<'a, A>> {
+    g.0
```

Comment:
this is rust for unpack newtype?

Reply:
Yeah they have the same form as a 1-tuple